test/lit/passes/signature-pruning.wast

```diff
@@ -393,12 +393,12 @@
 )
 
 (module
-  ;; CHECK:      (type $sig (func (param i32)))
+  ;; CHECK:      (type $sig (sub (func (param i32))))
   (type $sig (func_subtype (param i32) func))
 
   ;; As above, but now an import also uses this signature, which prevents us
   ;; from changing anything.
-  ;; CHECK:      (import "out" "func" (func $import (type $sig) (param i32)))
+  ;; CHECK:      (import "out" "func" (func $import (type $func.0) (param i32)))
```

Comment:
Not specific to this PR, but this automatic type picking has always felt surprising and dangerous to me...

Reply:
Oh yikes, this appears to be a real bug. `$func.0` does not exist.

Reply:
Fixed separately in https://github.com/WebAssembly/binaryen/pull/5927.